akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala

```diff
@@ -70,7 +71,17 @@ private[http] class HttpResponseParser(protected val settings: ParserSettings, p
         case 200 ⇒ StatusCodes.OK
         case code ⇒ StatusCodes.getForKey(code) match {
           case Some(x) ⇒ x
-          case None    ⇒ customStatusCodes(code) getOrElse badStatusCodeSpecific(code)
+          case None ⇒ customStatusCodes(code) getOrElse {
+            // A client must understand the class of any status code, as indicated by the first digit, and
+            // treat an unrecognized status code as being equivalent to the x00 status code of that class
+            // https://tools.ietf.org/html/rfc7231#section-6
+            try {
+              val reason = asciiString(input, reasonStartIdx, reasonEndIdx)
+              StatusCodes.custom(code, reason)
+            } catch {
+              case _: Exception ⇒ badStatusCodeSpecific(code)
```

Comment:
What kind of exceptions do we expect here?

Reply:
`IllegalArgumentException` will be thrown by `StatusCodes.custom()` if the parsed `code` is _not_ between 100 and 599. Maybe the code is overly protective here and should be changed to expect only this particular exception, not sure about that.

Reply:
In that case, maybe it would make sense to refactor it so that the `code` is checked explicitly before calling custom?

Reply:
Sure, can be done. My motivation to implement it this way was not to duplicate logic inside `StatusCodes.custom()` to `HttpResponseParser` (e.g. if we do that, `code` value will be checked twice, in both places). If this is not a concern, I can adjust.

Reply:
Ok, sounds good, thanks for the explanation!